api/src/transactions.rs

```diff
@@ -19,79 +19,45 @@ use diem_types::{
 
 use anyhow::Result;
 use warp::{
+    filters::BoxedFilter,
     http::{header::CONTENT_TYPE, StatusCode},
     reply, Filter, Rejection, Reply,
 };
 
-pub fn routes(context: Context) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
-    get_transaction(context.clone())
-        .or(get_transactions(context.clone()))
-        .or(get_account_transactions(context.clone()))
-        .or(submit_bcs_transactions(context.clone()))
-        .or(submit_json_transactions(context.clone()))
-        .or(create_signing_message(context))
-}
-
 // GET /transactions/{txn-hash / version}
-pub fn get_transaction(
-    context: Context,
-) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+pub fn get_transaction(context: Context) -> BoxedFilter<(impl Reply,)> {
     warp::path!("transactions" / TransactionIdParam)
         .and(warp::get())
         .and(context.filter())
         .and_then(handle_get_transaction)
         .with(metrics("get_transaction"))
-}
-
-async fn handle_get_transaction(
```

Comment:
moved handle_xx functions down, and group the routes definition functions on top